fenwick/src/main/java/org/opencadc/fenwick/IncludeArtifacts.java

```diff
@@ -0,0 +1,167 @@
+ */
+
+package org.opencadc.fenwick;
+
+import ca.nrc.cadc.util.StringUtil;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.log4j.Logger;
+
+
+public class IncludeArtifacts extends ArtifactSelector {
+    private static final Logger log = Logger.getLogger(IncludeArtifacts.class);
+
+    /**
+     * Empty constructor.
+     */
+    public IncludeArtifacts() {
+        super();
+    }
+
+    /**
+     * Obtain the iterator of clauses used to build a query to filter the Artifacts being merged.  This method will read
+     * in the files each time, so use it sparingly.
+     *
+     * @return Iterator of String clauses, or empty list.  Never null.
+     *
+     * @throws Exception Any issues with establishing the list, or reading from configuration
+     */
+    @Override
+    public Iterator<String> iterateIncludeClauses() throws Exception {
+        final List<String> whereClauses = new ArrayList<>();
+        loadClauses(whereClauses);
+        if (whereClauses.isEmpty()) {
+            throw new IllegalStateException(
+                    String.format("No usable SQL filter files located in %s.  Ensure there is at least one file "
+                                  + "with the .sql extension whose content begins with the keyword 'WHERE'.",
+                                  selectorConfigDir));
+        }
+        return whereClauses.iterator();
+    }
+
+    /**
+     * Iterate the files in the filters folder and read in the appropriately named files.  This method will verify that
+     * the file names end with an SQL extension, and the clause declared within begins with WHERE.  That clause is
+     * then loaded into the given list with the WHERE keyword stripped.
+     * <p/>
+     * This method is how the SQL files are read in and loaded as clauses.
+     *
+     * @param whereClauses A list of clauses to load into.
+     * @throws IOException If anything goes awry while reading the files.
+     */
+    void loadClauses(final List<String> whereClauses) throws IOException {
+        final File configurationDirFile = selectorConfigDir.toFile();
+        if (configurationDirFile.isDirectory() && configurationDirFile.canRead()) {
+            final File[] fileListing = Objects.requireNonNull(
+                    configurationDirFile.listFiles(pathname -> pathname.getName().toLowerCase().endsWith(".sql")));
+            for (final File f : fileListing) {
+                boolean isFirstLine = true;
+                final StringBuilder clauseBuilder = new StringBuilder();
+                // There may be multiple lines per file.  Trim each one and treat it like a single clause.
+                for (final String clause : Files.readAllLines(f.toPath())) {
+                    if (StringUtil.hasText(clause)) {
+                        // The first line MUST begin with the WHERE keyword.
```

Comment:
I would allow for comment lines in these files (lines starting with --) so I guess you have to read them with a LineNumberReader and discard comments and I'd probably sanitise white space (at least blank lines and maybe newlines as well). At a minimum, we could then have an example that worked with an explanation.

All that stuff currently works fine in our TAP service but I don't think it is strictly required by TAP and it makes it hard to find the WHERE.

Reply:
Done.